packages/mux-uploader/src/index.ts

```diff
@@ -0,0 +1,419 @@
+        // circumference of our circle) begins. The larger the offset, the farther into the circle you're
+        // starting the "dash". In the beginning, offset is the same as the circumference. Meaning, the visible
+        // dash starts at the end so we don't see the full circle. Instead we see a gap the size of the circle.
+        // When the percentage is 100%, offset is 0 meaning the dash starts at the beginning so we can see the circle. (TD).
+
+        this.svgCircle.style.strokeDasharray = `${this.circumference} ${this.circumference}`;
+        this.svgCircle.style.strokeDashoffset = `${this.circumference}`;
+      }
+    }
+  }
+
+  setupRetry() {
+    this.retryMessage?.addEventListener('click', () => {
+      this.removeAttribute('upload-error');
+      this.removeAttribute('upload-in-progress');
+      if (this.statusMessage) this.statusMessage.innerHTML = '';
+      if (this.uploadPercentage) this.uploadPercentage.innerHTML = '';
+    });
+  }
+
+  setupDropHandler() {
+    //@ts-ignore
+    this.addEventListener('mux-drop', this._dropHandler);
+  }
+
+  setupFilePickerButton() {
+    this.shadowRoot?.querySelector('slot[name=custom-button]')?.addEventListener('slotchange', () => {
+      this.filePickerButton = this.shadowRoot?.querySelector('slot[name=custom-button]');
+    });
+
+    this.filePickerButton?.addEventListener('click', () => {
+      // TO-DO: Allows user to reattempt uploading the same file after an error.
+      // Note: Apparently Chrome and Firefox do not allow changing an indexed property on FileList...(TD).
+      // Source: https://stackoverflow.com/a/46689013
+
+      this.hiddenFileInput?.click();
+    });
+
+    this.hiddenFileInput?.addEventListener('change', (evt) => {
+      const file = this.hiddenFileInput?.files && this.hiddenFileInput?.files[0];
+
+      if (file) {
+        this.dispatchEvent(
+          new CustomEvent('mux-drop', {
+            composed: true,
+            bubbles: true,
+            detail: file,
+          })
+        );
+      }
+    });
+  }
+
+  setProgress(percent: number) {
+    if (this.uploadPercentage) this.uploadPercentage.innerHTML = `${Math.floor(percent)}%`;
+    this.setAttribute('aria-valuenow', `${Math.floor(percent)}`);
+    this.setAttribute('aria-valuetext', `${Math.floor(percent)} percent`);
+
+    switch (this.getAttribute('type')) {
+      case TYPES.BAR: {
+        if (this.progressBar) this.progressBar.style.width = `${percent}%`;
+      }
+      case TYPES.RADIAL: {
+        if (this.svgCircle) {
+          // The closer the upload percentage gets to 100%, the closer offset gets to 0.
+          // The closer offset gets to 0, the more we can see the circumference of our circle. (TD).
+          const offset = this.circumference - (percent / 100) * this.circumference;
+
+          this.svgCircle.style.strokeDashoffset = offset.toString();
+        }
+      }
+    }
+  }
+
+  handleUpload(evt: CustomEvent) {
+    const url = this.getAttribute('url');
```

Comment:
(blocking): missing the `url` setter and getter, this seems like an attribute that should also be settable via an element property